image_op/reconstruct_traces.m

```diff
@@ -182,15 +185,33 @@ function reconstruct_traces(movie_source, ica_dir, varargin)
 
 traces = zeros(num_frames, rec_filter_count, 'single');
 fprintf('%s: Reconstructing traces...\n', datestr(now));
-for cell_idx = 1:rec_filter_count
-    rec_filter = filters(:,:,cell_idx);
-    pix_active = find(rec_filter>0);
-    movie_portion = M(pix_active,:)';
-    movie_portion(movie_portion<threshMov) = 0;
-    trace_this = (movie_portion * rec_filter(pix_active))';  
-    traces(:,cell_idx) = fix_baseline(trace_this);
+
+if use_legacy
+    % Inner product method (Not recommended)
+    for cell_idx = 1:rec_filter_count
+        rec_filter = filters(:,:,cell_idx);
+        pix_active = find(rec_filter>0);
+        movie_portion = M(pix_active,:)';
+        movie_portion(movie_portion<threshMov) = 0;
+        trace_this = (movie_portion * rec_filter(pix_active))';  
+        traces(:,cell_idx) = fix_baseline(trace_this);
+    end
+else
+    %Least Squares Method
+    F = reshape(filters,height*width,rec_filter_count);
+    idx_nonzero = find(sum(F,2)>0);
+    F_small = F(idx_nonzero,:); 
+    
+    % Caution: Below line requires memory of size roughly 10-20% of M
+    traces = (F_small'*F_small)\(F_small'*M(idx_nonzero,:)); 
+    traces = traces';
+    for k = 1:rec_filter_count;
+        traces(:,k) = fix_baseline(traces(:,k));
```

Comment:
You can pull `fix_baseline` out of the `use_legacy` if-then statement, since it's common to both branches